hono/src/services/items.ts

```diff
@@ -23,8 +23,50 @@ const selectItemSchema = createSelectSchema(items, {
   id: z.string().refine((val) => isCuid(val)),
 })
 
-app.get('/', injectDB, async (c) => {
-  return c.json(await c.get('db').query.items.findMany())
+const paginationValidationItems = z.object({
+  page: z
+    .string()
+    .refine((val) => !isNaN(+val) && +val >= 0, {
+      message: 'Items page number must be a non-negative number',
+    })
+    .optional(),
+  size: z
+    .string()
+    .refine((val) => !isNaN(+val) && +val > 0, {
+      message: 'Items page size must be a positive number',
+    })
+    .optional(),
+})
+
+app.get('/', zValidator('query', paginationValidationItems), injectDB, async (c) => {
+  const { page, size } = c.req.query()
+
+  const pageNumber: number = page ? +page : 0
+  const pageSize: number = size ? +size : 25
+
+  const itemsData = await c
+    .get('db')
+    .select()
+    .from(items)
+    .where(eq(items.authorUsername, 'jdoe'))
+    .limit(pageSize)
+    .offset(pageNumber)
+
+  const estimate = await c.get('db').execute(sql`
+      SELECT reltuples AS estimate
+      FROM pg_class
+      WHERE relname = 'items'
+        AND EXISTS (
+          SELECT 1 
+          FROM ${items}
+          WHERE ${items.authorUsername} = 'jdoe'
```

Comment:
Can we keep the hardcoded `authorUsername` as "rak3rman". @rak3rman is still using production Shafa on a daily basis and would like new items to map to "rak3rman" for now :)

Reply:
Yeah we can, but that also means I will need to change authorUsername for smoke tests to be "rak3erman" (if you are ok with that)

Reply:
Yes, that is ok!

Reply:
Resolved.